apps/src/applab/CrosshairOverlay.js

```diff
@@ -0,0 +1,144 @@
+/** @file Creates and controls a coordinates crosshair on the app visualization. */
+/* jshint
+ funcscope: true,
+ newcap: true,
+ nonew: true,
+ shadow: false,
+ unused: true,
+ eqeqeq: true,
+
+ maxlen: 90,
+ maxparams: 6,
+ maxstatements: 200
+ */
+'use strict';
+
+var constants = require('../constants');
+var SVG_NS = constants.SVG_NS;
+
+var CROSSHAIR_MARGIN = 6;
+var EDGE_MARGIN = 5;
+var TEXT_RECT_WIDTH = 104;
+var TEXT_RECT_HEIGHT = 21;
+var TEXT_RECT_RADIUS = TEXT_RECT_HEIGHT / 3;
+var TEXT_Y_OFFSET = -7;
+
+/**
+ * Creates and controls a coordinates crosshair on the app visualization.
+ * @constructor
+ */
+var CrosshairOverlay = function () {
+  /** @private {SVGGElement} */
+  this.ownElement_ = null;
+
+  /** @private {Object} */
+  this.props_ = {
+    x: 0,
+    y: 0,
+    appWidth: 0,
+    appHeight: 0
+  };
+};
+module.exports = CrosshairOverlay;
+
+/**
+ * @param {SVGElement} intoElement
+ * @param {Object} props
+ * @param {number} props.x
+ * @param {number} props.y
+ * @param {number} props.appWidth
+ * @param {number} props.appHeight
+ */
+CrosshairOverlay.prototype.render = function (intoElement, props) {
```

Comment:
I might call this nextProps instead of props for less confusion. There's some precedent for nextProps (i.e. see https://facebook.github.io/react/docs/component-specs.html).